test/fixtures/s3-fixture/src/main/java/fixture/s3/S3HttpHandler.java

```diff
@@ -207,8 +207,9 @@ public void handle(final HttpExchange exchange) throws IOException {
                 exchange.sendResponseHeaders(RestStatus.OK.getStatus(), -1);
 
             } else if (request.isListObjectsRequest()) {
-                if (request.queryParameters().containsKey("list-type")) {
-                    throw new AssertionError("Test must be adapted for GET Bucket (List Objects) Version 2");
+                final var listType = request.queryParameters().getOrDefault("list-type", List.of("1")).get(0);
```

Comment:
Could you clarify with a comment that V2 requests newly have this extra version (?) field, so we're adding the check in case the version unexpectedly changes in future? That's my understanding, anyway, from looking around on the internet.

Reply:
I think we can replace it something like 

```java
final var listType = request.getOptionalQueryParam("list-type");
assert listType.isEmpty() || "2".equals(listType.get());
```

Seems more explicit with the intention and also ensures the parameter has a single value?